service/controller/clusterapi/v29/resource/clusterazs/create.go

```diff
@@ -266,34 +283,45 @@ func fromEC2SubnetsToMap(ss []*ec2.Subnet) (map[string]subnetPair, error) {
 	return azMap, nil
 }
 
-func mapAZSubnetsToControllerContextTypes(azs map[string]subnetPair) []controllercontext.ContextStatusTenantClusterAvailabilityZone {
-	var results []controllercontext.ContextStatusTenantClusterAvailabilityZone
+func getAZStatus(ctx context.Context, azs map[string]subnetPair) []controllercontext.ContextStatusTenantClusterAvailabilityZone {
+	var azNames []string
+	{
+		for az := range azs {
+			azNames = append(azNames, az)
+		}
 
-	for az, subnet := range azs {
-		ccAZ := controllercontext.ContextStatusTenantClusterAvailabilityZone{
+		sort.Strings(azNames)
+	}
+
+	var statusAZs []controllercontext.ContextStatusTenantClusterAvailabilityZone
+
+	for _, az := range azNames {
+		subnets := azs[az]
```

Comment:
We should continue here when the subnets are empty. So not tracking AZs that do not yet have an allocated subnet. Because the allocation gets into the desired state if I get this right.